internal/bminventory/inventory.go

```diff
@@ -1510,6 +1511,69 @@ func (b *bareMetalInventory) GetClusterDefaultConfig(_ context.Context, _ instal
 	return installer.NewGetClusterDefaultConfigOK().WithPayload(&body)
 }
 
+func getPlatformFromVendor(vendor models.SystemVendor) models.PlatformType {
+	var supportedManufacturers = map[string]models.PlatformType{
+		"VMware, Inc.": models.PlatformTypeVsphere,
+	}
+	return supportedManufacturers[vendor.Manufacturer]
+}
+
+func decideSupportedPlatforms(hostsPlatforms []models.PlatformType, hostsCount int) *[]models.PlatformType {
+	defaultPlatform := models.PlatformTypeBaremetal
+	supportedPlatforms := []models.PlatformType{defaultPlatform}
+
+	// SNO or no hosts
+	if hostsCount == 0 || hostsCount == 1 {
+		return &supportedPlatforms
+	}
+
+	platforms := make(map[models.PlatformType]bool)
+	existsPlatformsSet := make([]models.PlatformType, 0)
```

Comment:
why do you need it? maybe instead you should use `supportedPlatforms`? 
Then you will not need the validation `len(existsPlatformsSet) != 1...`